newscoop/template_engine/classes/CampContext.php

```diff
@@ -1351,4 +1351,14 @@ public function getSystemPreference($preferenceName, $default = null)
 
         return $preferencesService->get($preferenceName, $default);
     }
+
+    /**
+    * Set preview mode for context.
+    *
+    * Setting preview mode allows rendering not published issues and articles.
+    */
+    public function setPreviewMode($status = true)
+    {
+        $this->m_readonlyProperties['preview'] = $status;
```

Comment:
Seems a bit strange to create a setter for member of an array called "readonlyProperties"